### ISSUE TITLE
When solving the MIP problem, can I set an initial feasible solution to HIGHS

### ISSUE THREAD

**Issue by tusiqi1** [ERGO-Code/HiGHS#1284]

**Comment by jajhall** [comment]
Yes, if you're calling HiGHS from an application you pass a solution to it before solving the model. If you're running it from the command line you use the

**Comment by tu3256076** [comment]
thanks for your answer, I have another question:
Can I turn off the "row number" check when use --read_solution_file? 
When I want to achieve multi-objective, because I know that the solution of the first model generates a feasible solution of the second model, but the second model has more constraints

**Comment by jajhall** [comment]
Interesting question. I'd not thought of this use case. As you suggest, the file can be modified by removing the rows section. However, this is a bit "ad hoc". I'd only suggested generating a HiGHS solution file so that you could see the format you need to generate independent of HiGHS. However, I can see you wanting to use HiGHS to generate the solution file each time. 

I guess that if the number of rows in the rows section doesn't match the number of rows in the model, I can just ignore the rows section. After all, it's not necessary when reading in a solution.

I'll modify the `latest` branch to do this

**Comment by jajhall** [comment]
Once #1286 is merged into `latest`, a solution file with a number of rows that differs from the incumbent model will be read. The row values in the file are discarded.

Note @tu3256076, if you add constraints to a MIP so that the saved solution is not feasible, then the MIP solver can only take advantage of the supplied solution if a feasible solution can be found (by solving for the continuous variables) for the supplied values of the integer variables.

**Comment by tu3256076** [comment]
Thank you for your support！

**Comment by tusiqi1** [comment]
How does highs1.6.0 sort cols in read lp files?I encountered the following problems:

1. Solving model1.lp:

Minimize
OBJROW: s(0,0) + s(0,1)
Subject To
constr(0):  s(0,0) >= 0
constr(1):  s(0,1) >= 0
eq7_00:  p(0,0) = -0
constr(13):  - s(0,0) + c(0,0) - p(0,0) = -0
eq7_01:  p(0,1) -20 x(0,1,30) -20 x(0,1,31) -20 x(0,1,32) -20 x(0,1,33) = -0
constr(15):  - s(0,1) + c(0,1) - p(0,1) = -0
eq9_0_0_0:  - epsilon_hat(0,0,0) = -0
eq10_0_0_0:  -10000000 x(0,0,0) + epsilon_bar(0,0,0) <= -0
eq11_left_0_0_0:  10000000 x(0,0,0) + epsilon_hat(0,0,0) - epsilon_bar(0,0,0) <= 10000000
eq11_right_0_0_0:  - epsilon_hat(0,0,0) + epsilon_bar(0,0,0) <= -0
eq9_0_1_30:  - epsilon_hat(0,1,30) = -1
eq10_0_1_30:  -10000000 x(0,1,30) + epsilon_bar(0,1,30) <= -0
eq11_left_0_1_30:  10000000 x(0,1,30) + epsilon_hat(0,1,30) - epsilon_bar(0,1,30) <= 10000000
eq11_right_0_1_30:  - epsilon_hat(0,1,30) + epsilon_bar(0,1,30) <= -0
eq9_0_1_31:  - epsilon_hat(0,1,31) = -1
eq10_0_1_31:  -10000000 x(0,1,31) + epsilon_bar(0,1,31) <= -0
eq11_left_0_1_31:  10000000 x(0,1,31) + epsilon_hat(0,1,31) - epsilon_bar(0,1,31) <= 10000000
eq11_right_0_1_31:  - epsilon_hat(0,1,31) + epsilon_bar(0,1,31) <= -0
eq9_0_1_32:  - epsilon_hat(0,1,32) = -1
eq10_0_1_32:  -10000000 x(0,1,32) + epsilon_bar(0,1,32) <= -0
eq11_left_0_1_32:  10000000 x(0,1,32) + epsilon_hat(0,1,32) - epsilon_bar(0,1,32) <= 10000000
eq11_right_0_1_32:  - epsilon_hat(0,1,32) + epsilon_bar(0,1,32) <= -0
eq9_0_1_33:  - epsilon_hat(0,1,33) = -1
eq10_0_1_33:  -10000000 x(0,1,33) + epsilon_bar(0,1,33) <= -0
eq11_left_0_1_33:  10000000 x(0,1,33) + epsilon_hat(0,1,33) - epsilon_bar(0,1,33) <= 10000000
eq11_right_0_1_33:  - epsilon_hat(0,1,33) + epsilon_bar(0,1,33) <= -0
constr(36):  epsilon(0,0) - epsilon_bar(0,0,0) = -0
constr(37):  epsilon(0,1) - epsilon_bar(0,1,30) - epsilon_bar(0,1,31) - epsilon_bar(0,1,32) - epsilon_bar(0,1,33) = -0
constr(38):  Cmax - c(0,1) >= -0
constr(39):  x(0,0,0) = 1
constr(40):  Cmax - c(0,1) >= -0
constr(41):  x(0,1,30) + x(0,1,31) + x(0,1,32) + x(0,1,33) = 1
constr(42):  - c(0,0) + s(0,1) >= -0
constr(43):  s(0,0) - epsilon(0,0) -10000000 x(0,0,0) >= -10000000
constr(44):  s(0,1) - epsilon(0,1) -10000000 x(0,1,30) >= -10000000
constr(45):  s(0,1) - epsilon(0,1) -10000000 x(0,1,31) >= -10000000
constr(46):  s(0,1) - epsilon(0,1) -10000000 x(0,1,32) >= -10000000
constr(47):  s(0,1) - epsilon(0,1) -10000000 x(0,1,33) >= -10000000
constr(48):  Cmax <= 21
Bounds
 0 <= x(0,0,0) <= 1
 0 <= x(0,1,30) <= 1
 0 <= x(0,1,31) <= 1
 0 <= x(0,1,32) <= 1
 0 <= x(0,1,33) <= 1
 0 <= help_eq4_u(0) <= 1
 0 <= help_eq4_v(0) <= 1
 0 <= help_eq4_u(30) <= 1
 0 <= help_eq4_v(30) <= 1
 0 <= help_eq4_u(31) <= 1
 0 <= help_eq4_v(31) <= 1
 0 <= help_eq4_u(32) <= 1
 0 <= help_eq4_v(32) <= 1
 0 <= help_eq4_u(33) <= 1
 0 <= help_eq4_v(33) <= 1
Integers
x(0,0,0) x(0,1,30) x(0,1,31) x(0,1,32) x(0,1,33) help_eq4_u(0) help_eq4_v(0) help_eq4_u(30) help_eq4_v(30) help_eq4_u(31) 
help_eq4_v(31) help_eq4_u(32) help_eq4_v(32) help_eq4_u(33) help_eq4_v(33) 
End
2.  get "model1.sol":
Model status
Optimal

# Primal solution values
Feasible
Objective 21
# Columns 34
Cmax 21
s(0,0) 0
s(0,1) 1
p(0,0) 0
c(0,0) 0
p(0,1) 20
x(0,1,30) 1
x(0,1,31) 0
x(0,1,32) 0
x(0,1,33) 0
c(0,1) 21
epsilon_hat(0,0,0) 0
x(0,0,0) 1
epsilon_bar(0,0,0) 0
epsilon_hat(0,1,30) 1
epsilon_bar(0,1,30) 1
epsilon_hat(0,1,31) 1
epsilon_bar(0,1,31) 0
epsilon_hat(0,1,32) 1
epsilon_bar(0,1,32) 0
epsilon_hat(0,1,33) 1
epsilon_bar(0,1,33) 0
epsilon(0,0) 0
epsilon(0,1) 1
help_eq4_u(0) 0
help_eq4_v(0) 0
help_eq4_u(30) 0
help_eq4_v(30) 0
help_eq4_u(31) 0
help_eq4_v(31) 0
help_eq4_u(32) 0
help_eq4_v(32) 0
help_eq4_u(33) 0
help_eq4_v(33) 0
# Rows 39
constr(0) 0
constr(1) 1
eq7_00 0
constr(13) 0
eq7_01 0
constr(15) 0
eq9_0_0_0 0
eq10_0_0_0 -10000000
eq11_left_0_0_0 10000000
eq11_right_0_0_0 0
eq9_0_1_30 -1
eq10_0_1_30 -9999999
eq11_left_0_1_30 10000000
eq11_right_0_1_30 0
eq9_0_1_31 -1
eq10_0_1_31 0
eq11_left_0_1_31 1
eq11_right_0_1_31 -1
eq9_0_1_32 -1
eq10_0_1_32 0
eq11_left_0_1_32 1
eq11_right_0_1_32 -1
eq9_0_1_33 -1
eq10_0_1_33 0
eq11_left_0_1_33 1
eq11_right_0_1_33 -1
constr(36) 0
constr(37) 0
constr(38) 0
constr(39) 1
constr(40) 0
constr(41) 1
constr(42) 1
constr(43) -10000000
constr(44) -10000000
constr(45) 0
constr(46) 0
constr(47) 0
constr(48) 21.0

# Dual solution values
None

# Basis
HiGHS v1
None

3. solve model2.lp, set read_solution_file=model1.sol, log is :
Running HiGHS 1.6.0: Copyright (c) 2023 HiGHS under MIT licence terms
MIP  model1 has 39 rows; 34 cols; 88 nonzeros; 15 integer variables
WARNING: Col      5 has         infeasibility of          19 from [lower, value, upper] = [              0;              20;               1]
WARNING: Row      0 has         residual      of          21
WARNING: Row      4 has         residual      of         420
WARNING: Row      7 has         residual      of       1e+07
WARNING: Row     11 has         residual      of     1.9e+08
Solution has               num          max          sum
Col     infeasibilities      1           19           19
Integer infeasibilities      0            0            0
Row     infeasibilities      0            0            0
Row     residuals           35      1.9e+08      6.3e+08
Attempting to find feasible solution for (partial) user-supplied values of discrete variables
Solving LP without presolve or with basis
Using EKK dual simplex solver - serial
  Iteration        Objective     Infeasibilities num(sum)
          0    -5.0455790979e-07 Pr: 9(2.30833e+06) 0s
          3     1.9000664003e+08 0s
Model   status      : Infeasible
Simplex   iterations: 3
Objective value     :  1.9000000000e+08
HiGHS run time      :          0.00

Presolve is switched off

MIP start solution is infeasible, objective value is 190000000

Solving MIP model with:
   39 rows
   34 cols (15 binary, 0 integer, 0 implied int., 19 continuous)
   88 nonzeros

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work      
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   0               inf                  inf        0      0      0         0     0.0s
 R       0       0         0   0.00%   0               1                100.00%        0      0      0         1     0.0s

Solving report
  Status            Optimal
  Primal bound      1
  Dual bound        1
  Gap               0% (tolerance: 0.01%)
  Solution status   feasible
                    1 (objective)
                    0 (bound viol.)
                    0 (int. viol.)
                    0 (row viol.)
  Timing            0.00 (total)
                    0.00 (presolve)
                    0.00 (postsolve)
  Nodes             1
  LP iterations     6 (total)
                    0 (strong br.)
                    5 (separation)
                    0 (heuristics)
Writing the solution to model1.sol

**Comment by tusiqi1** [comment]
The only difference between the two files is objective and constr(48). 
![image](https://github.com/ERGO-Code/HiGHS/assets/38119921/0907646c-9e4a-4a59-a0bf-45cded171b5d)
 
However,highs gives solutions for different order of independent variables.
![image](https://github.com/ERGO-Code/HiGHS/assets/38119921/016fa821-5c5b-4862-8c98-aea27e78239e)

This causes me to fail using the read_solution_file parameter

**Comment by jajhall** [comment]
HiGHS doesn't "sort" the columns, it just defines them in the order they are encountered in the .lp (or MPS) file. Hence in lp1, CMax is column 0, but in lp2 it's not encountered until later. Hence the solution file generated from lp1 can't be used as a starting solution for lp2. 

Do you know what Gurobi does in this case?

That said, it would be possible to read the solution file and match its data to the column names - if possible. I can't imagine anyone wanting to use a solution file based on the column order of one LP to start the solution of another LP with the same column names, but permuted, so nothing is lost. If the names aren't permuted then the solution data would be used in the order given.

I'll create an issue to remind me to implement this, but I won't be doing it soon.

In the meantime you'll just have to keep the same implied column order in your .lp files

**Comment by tusiqi1** [comment]
> HiGHS doesn't "sort" the columns, it just defines them in the order they are encountered in the .lp (or MPS) file. Hence in lp1, CMax is column 0, but in lp2 it's not encountered until later. Hence the solution file generated from lp1 can't be used as a starting solution for lp2.
> 
> Do you know what Gurobi does in this case?
> 
> That said, it would be possible to read the solution file and match its data to the column names - if possible. I can't imagine anyone wanting to use a solution file based on the column order of one LP to start the solution of another LP with the same column names, but permuted, so nothing is lost. If the names aren't permuted then the solution data would be used in the order given.
> 
> I'll create an issue to remind me to implement this, but I won't be doing it soon.
> 
> In the meantime you'll just have to keep the same implied column order in your .lp files

I'm not sure how gurobi did it.

I have made my program run correctly by making additional changes to the lp1 sol file order. 

In addition, I found that the results is not equal when "--presolve==off" for my lp1.

lp1 is:

`Minimize
OBJROW: Cmax
Subject To
constr(0):  s(0,0) >= 0
constr(1):  s(0,1) >= 0
eq7_00:  p(0,0) = -0
constr(13):  - s(0,0) + c(0,0) - p(0,0) = -0
eq7_01:  p(0,1) -20 x(0,1,30) -20 x(0,1,31) -20 x(0,1,32) -20 x(0,1,33) = -0
constr(15):  - s(0,1) + c(0,1) - p(0,1) = -0
eq9_0_0_0:  - epsilon_hat(0,0,0) = -0
eq10_0_0_0:  -10000000 x(0,0,0) + epsilon_bar(0,0,0) <= -0
eq11_left_0_0_0:  10000000 x(0,0,0) + epsilon_hat(0,0,0) - epsilon_bar(0,0,0) <= 10000000
eq11_right_0_0_0:  - epsilon_hat(0,0,0) + epsilon_bar(0,0,0) <= -0
eq9_0_1_30:  - epsilon_hat(0,1,30) = -1
eq10_0_1_30:  -10000000 x(0,1,30) + epsilon_bar(0,1,30) <= -0
eq11_left_0_1_30:  10000000 x(0,1,30) + epsilon_hat(0,1,30) - epsilon_bar(0,1,30) <= 10000000
eq11_right_0_1_30:  - epsilon_hat(0,1,30) + epsilon_bar(0,1,30) <= -0
eq9_0_1_31:  - epsilon_hat(0,1,31) = -1
eq10_0_1_31:  -10000000 x(0,1,31) + epsilon_bar(0,1,31) <= -0
eq11_left_0_1_31:  10000000 x(0,1,31) + epsilon_hat(0,1,31) - epsilon_bar(0,1,31) <= 10000000
eq11_right_0_1_31:  - epsilon_hat(0,1,31) + epsilon_bar(0,1,31) <= -0
eq9_0_1_32:  - epsilon_hat(0,1,32) = -1
eq10_0_1_32:  -10000000 x(0,1,32) + epsilon_bar(0,1,32) <= -0
eq11_left_0_1_32:  10000000 x(0,1,32) + epsilon_hat(0,1,32) - epsilon_bar(0,1,32) <= 10000000
eq11_right_0_1_32:  - epsilon_hat(0,1,32) + epsilon_bar(0,1,32) <= -0
eq9_0_1_33:  - epsilon_hat(0,1,33) = -1
eq10_0_1_33:  -10000000 x(0,1,33) + epsilon_bar(0,1,33) <= -0
eq11_left_0_1_33:  10000000 x(0,1,33) + epsilon_hat(0,1,33) - epsilon_bar(0,1,33) <= 10000000
eq11_right_0_1_33:  - epsilon_hat(0,1,33) + epsilon_bar(0,1,33) <= -0
constr(36):  epsilon(0,0) - epsilon_bar(0,0,0) = -0
constr(37):  epsilon(0,1) - epsilon_bar(0,1,30) - epsilon_bar(0,1,31) - epsilon_bar(0,1,32) - epsilon_bar(0,1,33) = -0
constr(38):  Cmax - c(0,1) >= -0
constr(39):  x(0,0,0) = 1
constr(40):  Cmax - c(0,1) >= -0
constr(41):  x(0,1,30) + x(0,1,31) + x(0,1,32) + x(0,1,33) = 1
constr(42):  - c(0,0) + s(0,1) >= -0
constr(43):  s(0,0) - epsilon(0,0) -10000000 x(0,0,0) >= -10000000
constr(44):  s(0,1) - epsilon(0,1) -10000000 x(0,1,30) >= -10000000
constr(45):  s(0,1) - epsilon(0,1) -10000000 x(0,1,31) >= -10000000
constr(46):  s(0,1) - epsilon(0,1) -10000000 x(0,1,32) >= -10000000
constr(47):  s(0,1) - epsilon(0,1) -10000000 x(0,1,33) >= -10000000
Bounds
 0 <= x(0,0,0) <= 1
 0 <= x(0,1,30) <= 1
 0 <= x(0,1,31) <= 1
 0 <= x(0,1,32) <= 1
 0 <= x(0,1,33) <= 1
 0 <= help_eq4_u(0) <= 1
 0 <= help_eq4_v(0) <= 1
 0 <= help_eq4_u(30) <= 1
 0 <= help_eq4_v(30) <= 1
 0 <= help_eq4_u(31) <= 1
 0 <= help_eq4_v(31) <= 1
 0 <= help_eq4_u(32) <= 1
 0 <= help_eq4_v(32) <= 1
 0 <= help_eq4_u(33) <= 1
 0 <= help_eq4_v(33) <= 1
Integers
x(0,0,0) x(0,1,30) x(0,1,31) x(0,1,32) x(0,1,33) help_eq4_u(0) help_eq4_v(0) help_eq4_u(30) help_eq4_v(30) help_eq4_u(31) 
help_eq4_v(31) help_eq4_u(32) help_eq4_v(32) help_eq4_u(33) help_eq4_v(33) 
End
`

**Comment by jajhall** [comment]
If the same objective value is obtained with and without presolve, that's all you can expect if your LP has a non-unique solution

**Comment by tusiqi1** [comment]
> > HiGHS doesn't "sort" the columns, it just defines them in the order they are encountered in the .lp (or MPS) file. Hence in lp1, CMax is column 0, but in lp2 it's not encountered until later. Hence the solution file generated from lp1 can't be used as a starting solution for lp2.
> > Do you know what Gurobi does in this case?
> > That said, it would be possible to read the solution file and match its data to the column names - if possible. I can't imagine anyone wanting to use a solution file based on the column order of one LP to start the solution of another LP with the same column names, but permuted, so nothing is lost. If the names aren't permuted then the solution data would be used in the order given.
> > I'll create an issue to remind me to implement this, but I won't be doing it soon.
> > In the meantime you'll just have to keep the same implied column order in your .lp files
> 
> I'm not sure how gurobi did it.
> 
> I have made my program run correctly by making additional changes to the lp1 sol file order.
> 
> In addition, I found that the results is not equal when "--presolve==off" for my lp1(highs 1.6.0).
> 
> lp1 is:
> 
> ** Minimize OBJROW: Cmax Subject To constr(0): s(0,0) >= 0 constr(1): s(0,1) >= 0 eq7_00: p(0,0) = -0 constr(13): - s(0,0) + c(0,0) - p(0,0) = -0 eq7_01: p(0,1) -20 x(0,1,30) -20 x(0,1,31) -20 x(0,1,32) -20 x(0,1,33) = -0 constr(15): - s(0,1) + c(0,1) - p(0,1) = -0 eq9_0_0_0: - epsilon_hat(0,0,0) = -0 eq10_0_0_0: -10000000 x(0,0,0) + epsilon_bar(0,0,0) <= -0 eq11_left_0_0_0: 10000000 x(0,0,0) + epsilon_hat(0,0,0) - epsilon_bar(0,0,0) <= 10000000 eq11_right_0_0_0: - epsilon_hat(0,0,0) + epsilon_bar(0,0,0) <= -0 eq9_0_1_30: - epsilon_hat(0,1,30) = -1 eq10_0_1_30: -10000000 x(0,1,30) + epsilon_bar(0,1,30) <= -0 eq11_left_0_1_30: 10000000 x(0,1,30) + epsilon_hat(0,1,30) - epsilon_bar(0,1,30) <= 10000000 eq11_right_0_1_30: - epsilon_hat(0,1,30) + epsilon_bar(0,1,30) <= -0 eq9_0_1_31: - epsilon_hat(0,1,31) = -1 eq10_0_1_31: -10000000 x(0,1,31) + epsilon_bar(0,1,31) <= -0 eq11_left_0_1_31: 10000000 x(0,1,31) + epsilon_hat(0,1,31) - epsilon_bar(0,1,31) <= 10000000 eq11_right_0_1_31: - epsilon_hat(0,1,31) + epsilon_bar(0,1,31) <= -0 eq9_0_1_32: - epsilon_hat(0,1,32) = -1 eq10_0_1_32: -10000000 x(0,1,32) + epsilon_bar(0,1,32) <= -0 eq11_left_0_1_32: 10000000 x(0,1,32) + epsilon_hat(0,1,32) - epsilon_bar(0,1,32) <= 10000000 eq11_right_0_1_32: - epsilon_hat(0,1,32) + epsilon_bar(0,1,32) <= -0 eq9_0_1_33: - epsilon_hat(0,1,33) = -1 eq10_0_1_33: -10000000 x(0,1,33) + epsilon_bar(0,1,33) <= -0 eq11_left_0_1_33: 10000000 x(0,1,33) + epsilon_hat(0,1,33) - epsilon_bar(0,1,33) <= 10000000 eq11_right_0_1_33: - epsilon_hat(0,1,33) + epsilon_bar(0,1,33) <= -0 constr(36): epsilon(0,0) - epsilon_bar(0,0,0) = -0 constr(37): epsilon(0,1) - epsilon_bar(0,1,30) - epsilon_bar(0,1,31) - epsilon_bar(0,1,32) - epsilon_bar(0,1,33) = -0 constr(38): Cmax - c(0,1) >= -0 constr(39): x(0,0,0) = 1 constr(40): Cmax - c(0,1) >= -0 constr(41): x(0,1,30) + x(0,1,31) + x(0,1,32) + x(0,1,33) = 1 constr(42): - c(0,0) + s(0,1) >= -0 constr(43): s(0,0) - epsilon(0,0) -10000000 x(0,0,0) >= -10000000 constr(44): s(0,1) - epsilon(0,1) -10000000 x(0,1,30) >= -10000000 constr(45): s(0,1) - epsilon(0,1) -10000000 x(0,1,31) >= -10000000 constr(46): s(0,1) - epsilon(0,1) -10000000 x(0,1,32) >= -10000000 constr(47): s(0,1) - epsilon(0,1) -10000000 x(0,1,33) >= -10000000 Bounds 0 <= x(0,0,0) <= 1 0 <= x(0,1,30) <= 1 0 <= x(0,1,31) <= 1 0 <= x(0,1,32) <= 1 0 <= x(0,1,33) <= 1 0 <= help_eq4_u(0) <= 1 0 <= help_eq4_v(0) <= 1 0 <= help_eq4_u(30) <= 1 0 <= help_eq4_v(30) <= 1 0 <= help_eq4_u(31) <= 1 0 <= help_eq4_v(31) <= 1 0 <= help_eq4_u(32) <= 1 0 <= help_eq4_v(32) <= 1 0 <= help_eq4_u(33) <= 1 0 <= help_eq4_v(33) <= 1 Integers x(0,0,0) x(0,1,30) x(0,1,31) x(0,1,32) x(0,1,33) help_eq4_u(0) help_eq4_v(0) help_eq4_u(30) help_eq4_v(30) help_eq4_u(31) help_eq4_v(31) help_eq4_u(32) help_eq4_v(32) help_eq4_u(33) help_eq4_v(33) End **

Highs 1.6.0 return different objective value for this lp file if without presolve. 
21(right) for presolve on and 20(wrong) for presolve off. I think highs may have bugs in this lp file.

**Comment by fwesselm** [comment]
```
Coefficient statistics:
  Matrix range [1e+00, 1e+07]
  Cost range   [1e+00, 1e+00]
  Bound range  [1e+00, 1e+00]
  RHS range    [1e+00, 1e+07]
```
I can reproduce the different answer with and without presolve. Maybe this is caused by the scaling of the inputs.

**Comment by jajhall** [comment]
Is the objective value the same in both cases?

**Comment by fwesselm** [comment]
Sorry, output WITH presolve:

```
Running HiGHS 1.6.0: Copyright (c) 2023 HiGHS under MIT licence terms
Coefficient statistics:
  Matrix range [1e+00, 1e+07]
  Cost range   [1e+00, 1e+00]
  Bound range  [1e+00, 1e+00]
  RHS range    [1e+00, 1e+07]
Presolving model
20 rows, 13 cols, 50 nonzeros  0.089522s
15 rows, 10 cols, 46 nonzeros  0.114762s
12 rows, 8 cols, 40 nonzeros  0.140895s
8 rows, 6 cols, 32 nonzeros  0.159539s
5 rows, 5 cols, 24 nonzeros  0.178468s
Objective function is integral with scale 1

Solving MIP model with:
   5 rows
   5 cols (3 binary, 0 integer, 2 implied int., 0 continuous)
   24 nonzeros

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work     
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   20              inf                  inf        0      0      0         0     0.2s
 R       0       0         0   0.00%   20              21                 4.76%        0      0      0         2     0.2s

Solving report
  Status            Optimal
  Primal bound      21
  Dual bound        21
  Gap               0% (tolerance: 0.01%)
  Solution status   feasible
                    21 (objective)
                    0 (bound viol.)
                    0 (int. viol.)
                    0 (row viol.)
  Timing            0.23 (total)
                    0.18 (presolve)
                    0.00 (postsolve)
  Nodes             1
  LP iterations     2 (total)
                    0 (strong br.)
                    0 (separation)
                    0 (heuristics)
```

Output WITHOUT presolve:

```
Running HiGHS 1.6.0: Copyright (c) 2023 HiGHS under MIT licence terms
Coefficient statistics:
  Matrix range [1e+00, 1e+07]
  Cost range   [1e+00, 1e+00]
  Bound range  [1e+00, 1e+00]
  RHS range    [1e+00, 1e+07]

Presolve is switched off

Solving MIP model with:
   38 rows
   34 cols (15 binary, 0 integer, 0 implied int., 19 continuous)
   87 nonzeros

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work     
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   0               inf                  inf        0      0      0         0     0.0s
 T       0       0         0   0.00%   0               20               100.00%        0      0      0         1     0.2s

Solving report
  Status            Optimal
  Primal bound      20
  Dual bound        20
  Gap               0% (tolerance: 0.01%)
  Solution status   feasible
                    20 (objective)
                    0 (bound viol.)
                    1.00000000058e-07 (int. viol.)
                    0 (row viol.)
  Timing            0.18 (total)
                    0.00 (presolve)
                    0.00 (postsolve)
  Nodes             1
  LP iterations     1 (total)
                    0 (strong br.)
                    0 (separation)
                    0 (heuristics)

```

**Comment by jajhall** [comment]
OK I'll investigate

**Comment by jajhall** [comment]
> Sorry, output WITH presolve:

At first I missed the comment by @tusiqi1 about the differing objective values. Shouldn't study issues on my phone when walking the dog!
